### DISCUSSION TITLE
[Bug] Graphql V4 languages missing size

### DISCUSSION THREAD

**Discussion by chmelevskij** [community/community#24478] [Apps, API and Webhooks]
<p>Following query gives me the total size in all the languages, which is not really useful.</p>
<pre><code>query {
  repository(owner:"facebook" name:"react") {
    languages(first:50) {
      totalSize #The total size in bytes of files written in that language. nodes {
        name
        color
      }
    }
  }
}
</code></pre>
<p>But what I’m looking for is something like </p>
<pre><code>query {
  repository(owner:"facebook" name:"react") {
    languages(first:50) {
      nodes {
        totalSize
        name
        color
      }
    }
  }
}
</code></pre>
<p>Expecting something similar to quering<br>
<a href="https://api.github.com/repos/facebook/react/languages" rel="nofollow noopener">https://api.github.com/repos/facebook/react/languages</a></p>
<p>Is this is a bug or am I missing something here?</p>

**Comment by nickvanw** [comment, marked as answer]
<p>Hey <a class="mention" href="/chmelevskij">@chmelevskij</a>!</p>
<p>I believe that it is possible to do what you want! Looking at the schema, it looks like the connection between Repositories and Languages does have an “edge type” that includes the number of bytes written in that language in the particular repository. Here’s the query that I’m using for Facebook’s React Repository:</p>
<pre><code>{
  repository(owner: "facebook", name: "react") {
    languages(first: 50, orderBy: {field: SIZE, direction: DESC}) {
      edges {
        size
        node {
          name
        }
      }
    }
  }
}
</code></pre>
<p>This gives me the same results as I can see in the REST API, sorted by the number of bytes in each language in descending order. </p>
<p>Let me know if I’ve misread, and have a great day!</p>
